typescript-typing-tutor/main.ts

```diff
@@ -1,38 +1,31 @@
-document.addEventListener('DOMContentLoaded', () => {
-  const spans = document.querySelectorAll('#phrase span');
-  let currentIndex = 0;
+const spans = document.querySelectorAll('#phrase span');
+let currentIndex = 0;
 
-  function updateCurrentSpan(): void {
-    spans.forEach((span, index): void => {
-      if (index === currentIndex) {
-        span.classList.add('current');
-      } else {
-        span.classList.remove('current');
-      }
-    });
-  }
-
-  function handleKeydown(event: KeyboardEvent): void {
-    if (currentIndex >= spans.length) return;
-    const currentSpan = spans[currentIndex];
-    const expectedChar = currentSpan.textContent;
-    const typedChar = event.key;
+function handleKeydown(event: KeyboardEvent): void {
+  if (currentIndex >= spans.length) return;
+  const currentSpan = spans[currentIndex];
+  const expectedChar = currentSpan.textContent;
+  const typedChar = event.key;
 
-    if (typedChar === expectedChar) {
-      currentSpan.classList.add('correct');
-      currentIndex++;
-    } else {
-      currentSpan.classList.add('incorrect');
-    }
+  if (currentIndex > 0) {
+    spans[currentIndex - 1].classList.remove('current');
+  }
 
-    setTimeout(() => {
-      currentSpan.classList.remove('incorrect');
-      updateCurrentSpan();
-    }, 500);
+  if (typedChar === expectedChar) {
+    currentSpan.classList.add('correct');
+    currentSpan.classList.remove('incorrect');
+    currentIndex++;
+  } else {
+    currentSpan.classList.add('incorrect');
+  }
 
-    updateCurrentSpan();
+  if (currentIndex < spans.length) {
+    spans[currentIndex].classList.add('current');
   }
+}
+
+document.addEventListener('keydown', handleKeydown);
 
-  document.addEventListener('keydown', handleKeydown);
-  updateCurrentSpan();
-});
+if (spans.length > 0) {
+  spans[0].classList.add('current');
+}
```

Comment:
This shouldn't be necessary since you added it in your HTML but you can leave this.